src/audio/asrc/asrc.c

```diff
@@ -29,20 +29,17 @@
 #define trace_asrc(__e, ...)					\
 	trace_event(TRACE_CLASS_SRC, __e, ##__VA_ARGS__)
 #define trace_asrc_with_ids(comp_ptr, __e, ...)			\
-	trace_event_comp(TRACE_CLASS_SRC, comp_ptr,		\
-			 __e, ##__VA_ARGS__)
+	trace_event_comp(SRC, comp_ptr, __e, ##__VA_ARGS__)
```

Comment:
Sorry, but I'm actually not in favour of this change (and others like this throughout the patch. The reasons are very simple: (1) it makes it more difficult to grep for these symbols, e.g. if you want to find all uses of `TRACE_CLASS_SRC` you won't find this any more. Also it makes following symbols in editors more difficult. (2) it makes it look different from the surrounding lines, e.g. the line above, which uses `TRACE_CLASS_SRC` in full.
If (1) is a trade-off - you win code compactness but you lose clarity, I think (2) is really not good - either convert all of them, or leave them all as is, don't mix them.

Reply:
Using "TRACE_CLASS_SRC" directly will not work correctly as that will be expanded to a number and will break the very idea of this PR, to disable verbose traces for particular classes. Maybe TRACE_CLASS_SRC with some prefix or suffix?

Reply:
@lyakh It was already picked up by @paulstelian97  and explained on other occurence of this change, here: https://github.com/thesofproject/sof/pull/2133#discussion_r349872229

The intermediate macro `trace_event_comp`, causes premature `TRACE_CLASS_*` token expansion, which could be solved by quite some dirty work under the hood in `trace_event` macro family expansions, or (in my opinion) more elegantly, here.
One could argue that greping for `TRACE_CLASS_*` doesn't make much sense anymore, as we've already wrapped (almost, reaching 100% soon) all traces in their respective `trace_*` macros.

If you have an idea how to solve token pasting and expansion more elegantly, I'm open for suggestions, but I honestly think that this route is cleaner than appending `_` to tokens to prevent their expansion, and then additionally accounting for it in `trace-verbose-private.h`.

Reply:
@paulstelian97 thanks for the explanation, I see now. But now my question is then: is it really worth it? Is it really worth adding all those levels of macros if we anyway have the trace-verbose-private.h header, which has a bunch of defines for each of the `CONFIG_TRACE_VERBOSE_*` options? Wouldn't it be much easier and compact to just add
```
#if CONFIG_TRACE_VERBOSE_xxx
#define trace_asrc_with_ids(comp_ptr, __e, ...) ...
#else
#define trace_asrc_with_ids(...) do {} while (0)
#endif
```
in every affected file? Most of them would then just involve changing a couple of lines at the top, others, calling generic macros directly from the code, like dai.c, without wrapping them, will need such a define added at the top, which would (arguably) also simplify them?

Reply:
@lyakh Tbh if we're doing that I'd say having the definition for `trace_asrc_with_ids` and every other such macro in a common header and make those macros public would be an easy way out. That sure is a large redo of this change and everyone would thus have to use these custom functions instead of the raw trace_event functions to benefit from that disabling.

Also in case we go with that, use the UNUSED macro instead of a do-while-0 loop. It was built specifically for using in disabled traces.

Reply:
@paulstelian97 I'd be fine keeping those defines in respective .c as long as they're used only there. I don't see a need for a common header if all or at least most of them are only used in a single file. However, if sufficiently many of them are or can be used in multiple files, then yes, we could use a common header.